From 06352517f7e88fee2e4933e68e679e7901ee70ed Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:28:05 -0400
Subject: [pci] Remove unused pcie_aspm_enabled()

Message-id: <20140719042804.19438.98073.stgit@gir.stowe>
Patchwork-id: 86013
O-Subject: [RHEL7.1 PATCH 51/81] PCI: Remove unused pcie_aspm_enabled()
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: 8f92fb06ffb090f2b22717481352f1e58511c834

commit 8f92fb06ffb090f2b22717481352f1e58511c834
Author: Stephen Hemminger <stephen@networkplumber.org>
Date:   Fri Jan 10 15:23:16 2014 -0700

    PCI: Remove unused pcie_aspm_enabled()

    My philosophy is unused code is dead code.  And dead code is subject to bit
    rot and is a likely source of bugs.  Use it or lose it.

    This reverts part of 3e1b16002af2 ("ACPI/PCI: PCIe ASPM _OSC support
    capabilities called when root bridge added"), removing this interface:

        pcie_aspm_enabled()

    [bhelgaas: split to separate patch]
    Signed-off-by: Stephen Hemminger <stephen@networkplumber.org>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    CC: Andrew Patterson <andrew.patterson@hp.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/pcie/aspm.c b/drivers/pci/pcie/aspm.c
index f1272dc..e1e7026 100644
--- a/drivers/pci/pcie/aspm.c
+++ b/drivers/pci/pcie/aspm.c
@@ -984,18 +984,6 @@ void pcie_no_aspm(void)
  }
 }
 
-/**
- * pcie_aspm_enabled - is PCIe ASPM enabled?
- *
- * Returns true if ASPM has not been disabled by the command-line option
- * pcie_aspm=off.
- **/
-int pcie_aspm_enabled(void)
-{
-       return !aspm_disabled;
-}
-EXPORT_SYMBOL(pcie_aspm_enabled);
-
 bool pcie_aspm_support_enabled(void)
 {
  return aspm_support_enabled;
diff --git a/include/linux/pci.h b/include/linux/pci.h
index b85a88a..99f26c5 100644
--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -1257,10 +1257,8 @@ extern bool pcie_ports_auto;
 #endif
 
 #ifndef CONFIG_PCIEASPM
-static inline int pcie_aspm_enabled(void) { return 0; }
 static inline bool pcie_aspm_support_enabled(void) { return false; }
 #else
-int pcie_aspm_enabled(void);
 bool pcie_aspm_support_enabled(void);
 #endif
 
-- 
1.7.1